src/Client/HttpExceptionHandler.php

```diff
@@ -73,7 +73,10 @@ public function transformResponseToException(RequestInterface $request, Response
      */
     protected function getResponseMessage(ResponseInterface $response)
     {
-        $decodedBody = json_decode($response->getBody()->getContents(), true);
+        $responseBody = $response->getBody();
+        $decodedBody = json_decode($responseBody->getContents(), true);
+
+        $responseBody->rewind();
```

Comment:
So it should be done before AND after. The API users can get the response from the exception, and  they surely don't think to do a rewind if they want to read the body.